### PR TITLE
Plugin manager

### PR DESCRIPTION
**Pull request by jcampbell05**
Adds a message when trying to invoke a previously bundled action from a Fastfile or using the one-off command.

The user is presented with a message informing that we no longer bundle it and to install it using the new plugin commands.

screen shot 2016-05-22 at 23 36 16

### CODE REVIEW

**Comment by jcampbell05** [comment]
@KrauseFx  As part of this should we shortlist actions to consider unbundling in the future ?

File: fastlane/lib/fastlane/one_off.rb [hunk 1]
```diff
@@ -29,7 +29,13 @@ def self.run(action: nil, parameters: nil)
       begin
         class_ref = Fastlane::Actions.const_get(class_name)
       rescue NameError
-        UI.user_error!("Action '#{action}' not available, run `fastlane actions` to get a full list")
+        if Fastlane::Actions.formerly_bundled_actions.include?(action)
+          # This was a formerly bundled action which is now a plugin.
+          UI.verbose(caller.join("\n"))
+          UI.user_error!("The action '#{action}' is no longer bundled with Fastlane. You can use it via the new Plugin system by running 'fastlane add_plugin #{action}'.")
```
Comment (KrauseFx) [inline fastlane/lib/fastlane/one_off.rb]: fastlane should be lower case

Comment (KrauseFx) [inline fastlane/lib/fastlane/one_off.rb]: Plugin system also lower case I guess

Comment (KrauseFx) [inline fastlane/lib/fastlane/one_off.rb]: I'd propose
```ruby
UI.user_error!("The action '#{action}' is no longer bundled with fastlane. You can install it using `fastlane add_plugin #{action}`")
```

File: fastlane/lib/fastlane/runner.rb [hunk 2]
```diff
@@ -99,7 +99,7 @@ def trigger_action_by_name(method_sym, custom_dir, *arguments)
       end
 
       # It's important to *not* have this code inside the rescue block
-      # otherwise all NameErrors will be caugth and the error message is
+      # otherwise all NameErrors will be caught and the error message is
```
Comment (KrauseFx) [inline fastlane/lib/fastlane/runner.rb]: 😷

File: fastlane/lib/fastlane/runner.rb [hunk 3]
```diff
@@ -119,6 +119,11 @@ def trigger_action_by_name(method_sym, custom_dir, *arguments)
             # That's a plugin, but for some reason we can't find it
             # TODO: Add actual link
             UI.user_error!("Plugin '#{method_sym}' was not properly loaded, make sure to follow the plugin docs for troubleshooting")
+
+          elsif Fastlane::Actions.formerly_bundled_actions.include?(method_str)
+            # This was a formerly bundled action which is now a plugin.
+            UI.verbose(caller.join("\n"))
+            UI.user_error!("The action '#{method_sym}' is no longer bundled with Fastlane. You can use it via the new Plugin system by running 'fastlane add_plugin #{method_sym}'.")
```
Comment (KrauseFx) [inline fastlane/lib/fastlane/runner.rb]: Same message as https://github.com/fastlane/fastlane/pull/4776/files#r64160177

**Comment by KrauseFx** [comment]
This is a great change, thanks @jcampbell05. I added some comments before it's ready to be merged. 

We don't plan on suddenly removing a high number of actions, but slowly deprecate more and more actions. It would be great to have `xcake` as one of the first publicly available plugins 👍 
We'll let you know in the #plugins channel when the `fastlane new_plugin` action is ready 🚀

**Comment by KrauseFx** [comment]
I'd also like some tests to cover this change 👍

**Comment by jcampbell05** [comment]
@KrauseFx do you have an example on a good way to test this output ?

**Comment by KrauseFx** [comment]
Looks great, thanks. I'll add some tests for you, I'll link you to the commit once it's ready 👍

**Comment by KrauseFx** [comment]
Here are the tests https://github.com/fastlane/fastlane/pull/4722/commits/a076f1503ba8c321cf0d4bf8c5ffbb940280bac4 🚀

**Pull request merged**